solidity/contracts/system/KeepFactorySelection.sol

```diff
@@ -191,11 +271,52 @@ library KeepFactorySelection {
             address(_self.factorySelector) == address(0),
             "Factory selector already set"
         );
+        require(address(_factorySelector) != address(0), "Invalid address");
+
+        _self.factorySelector = KeepFactorySelector(_factorySelector);
+    }
+
+    /// @notice Locks versions of factories. When lock is set vendor contracts
+    /// won't be called anymore to obtain the latest factories versions.
+    /// It requires expected factories addresses to be provided to protect from
+    /// locking on unexpected addresses.
```

Comment:
We should make it clear in docs this function can be called only one time and that the version lock can not be later disabled.